### PR TITLE
Add support for multiple contexts

### PR DESCRIPTION
**Pull request by AyoubElk**
To do:
- [x] Add tests
- [x] Add documentation section that explains how to navigate between different contexts (either using Link components with absolute paths, or directly using an `<a>` tag)

### CODE REVIEW

File: packages/fastify-renderer/src/node/renderers/Renderer.ts [hunk 1]
```diff
@@ -9,6 +9,13 @@ export interface RenderOptions {
   base: string
 }
 
+export type PartialRenderOptions =
```
Comment (airhorns) [inline packages/fastify-renderer/src/node/renderers/Renderer.ts:12]: Could you use `Partial<RenderOptions>` here (or in the place where this type is used)?

**Comment by airhorns** [comment]
This approach makes sense to me! A lot of thinking for a small diff, classic :)

File: packages/fastify-renderer/src/node/renderers/Renderer.ts (hunk 1 above)
Comment (AyoubElk) [reply packages/fastify-renderer/src/node/renderers/Renderer.ts:12]: My understanding of `Partial` is that it makes all properties optional and doesn't allow specifying dependencies between the fields (i.e. if the `layout` property is present then the `base` property is required).

**Review by airhorns — APPROVED** [review]
LGTM! Nice work thinking it through!

**Pull request merged**